modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1522,16 +1522,48 @@ private void _validateListTypeEntriesValues(
 		}
 	}
 
+	private void _validateObjectFieldIntegerTypeSize(
+			Map.Entry<String, Serializable> entry)
+		throws ObjectEntryValuesException {
+
+		Serializable entryValue = entry.getValue();
+
+		if (!StringUtil.equals(entryValue.toString(), "")) {
+			int value = GetterUtil.getInteger(entryValue);
+
+			if (!StringUtil.equals(
+					String.valueOf(value), entryValue.toString())) {
+
+				throw new ObjectEntryValuesException.IntegerMaxSizeExceeded();
+			}
+		}
+	}
+
+	private void _validateObjectFieldLongType(
```

Comment:
Hey @mateussgn , It would be nice to avoid the duplication code between these 2 methods (_validateObjectFieldIntegerTypeSize and _validateObjectFieldLongType)